File: Mid Number/Mid_Number.c
#include<errno.h>
#include<limits.h>
#include<stdlib.h>
int main(int argc, char** argv)
{
    char *p;
    int num;
    errno = 0;
    long conv = strtol(argv[1], &p, 10);
    if (errno != 0 || *p != '\0' || conv > INT_MAX)
    {
        printf("Input error!");
    }
    else {
        num = conv;
        int digit = num%10;
        num = num/10;
        digit = num%10;
        if(digit == 2 || digit == 3 || digit == 5 || digit == 7)
            printf("prime");
        else
            printf("not prime");
    }
}

File: Mid Number/Mid_Number.java
class Mid_Number {
    public static void main(String[] args) {
        int num = Integer.parseInt(args[0]);
        if(args.length==1){
            int digit = num % 10;
            num = num / 10;
            digit = num % 10;
            if(digit==2 || digit==3 || digit==5 || digit==7)
                System.out.println("Prime");
            else
                System.out.println("Not Prime");
        }
        else if(args.length>1)
            System.out.println("Multiple Input Detected");
        else
            System.out.println("No Input Detected : Something Went Wrong");

    }
}